web-common/src/features/themes/color-utils.ts

```diff
@@ -0,0 +1,78 @@
+import type { V1Color } from "@rilldata/web-common/runtime-client";
+
+export type ThemeColor = [number, number, number];
+
+export function convertColor(color: V1Color): ThemeColor {
+  return [color.red ?? 0, color.green ?? 0, color.blue ?? 0].map(
+    (c) => c * 256
+  ) as ThemeColor;
+}
+
+export function RGBToHSL(rgb: ThemeColor): ThemeColor {
+  const r = rgb[0] / 255;
+  const g = rgb[1] / 255;
+  const b = rgb[2] / 255;
+  const min = Math.min(r, g, b);
+  const max = Math.max(r, g, b);
+  const delta = max - min;
+  let h: number;
+  let s: number;
+
+  if (max === min) {
+    h = 0;
+  } else if (r === max) {
+    h = (g - b) / delta;
+  } else if (g === max) {
+    h = 2 + (b - r) / delta;
+  } else if (b === max) {
+    h = 4 + (r - g) / delta;
+  }
+
+  h = Math.min(h * 60, 360);
```

Comment:
I'm seeing this error in my editor:
![image](https://github.com/rilldata/rill/assets/14206386/63cc97b8-0c6b-43b1-bf84-a1292d7b2120)